batch/test/test_batch.py

```diff
@@ -704,3 +704,35 @@ def approx_equal(x, y, tolerance=1e-10):
             assert approx_equal(cost_by_batch, cost_by_billing_project), (cost_by_batch, cost_by_billing_project)
         finally:
             client.close()
+
+    def test_billing_limit_zero(self):
+        client = BatchClient('test-zero-limit')
+        try:
+            batch = client.create_batch()
+            batch = batch.submit()
+        except aiohttp.ClientResponseError as e:
+            assert e.status == 403 and 'has exceeded the budget' in e.message
+        else:
+            assert False, f'should receive a 403 Forbidden {batch.id}'
+        finally:
+            client.close()
+
+    def test_billing_limit_tiny(self):
+        client = BatchClient('test-tiny-limit')
+        try:
+            batch = client.create_batch()
+            j1 = batch.create_job('ubuntu:18.04', command=['sleep', '5'])
+            j2 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j1])
+            j3 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j2])
+            j4 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j3])
+            j5 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j4])
+            j6 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j5])
+            j7 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j6])
+            j8 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j7])
+            j9 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j8])
+            j10 = batch.create_job('ubuntu:18.04', command=['sleep', '5'], parents=[j9])
+            batch = batch.submit()
+            batch = batch.wait()
+            assert batch['state'] == 'cancelled', batch
```

Comment:
I feel confident this test will fail, but I want something even stronger. If the cancel test runs every ten seconds, then I shouldn't overspend by, say, more than 30 seconds? Or at least one minute?
```python3
try:
    batch = client.create_batch()
    j1 = batch.create_job('ubuntu:18.04', command=['sleep', '5'])
    j2 = batch.create_job('ubuntu:18.04',
                          command=['/bin/bash', '-c', 'while true; do; sleep 3600; done'],
                          parents=[j1])
    batch = batch.submit()
    status = batch.wait()
    assert status['state'] == 'cancelled', batch
    jobs = {b['job_id']: b for b in batch.jobs()}
    assert jobs[2]['msec_mcpu'] < 30_000 * 125  # 30s on 125mCPU
```
Hmm. Oddly we don't have durations for cancelled jobs?